hera_cal/abscal.py

```diff
@@ -992,6 +992,73 @@ def RFI_delay_slope_cal(reds, antpos, red_data, freqs, rfi_chans, rfi_headings,
         gains[ant] = np.exp(2j * np.pi * np.outer(dlys, freqs))
     return gains
 
+def cross_pol_phase_cal(model, data, model_bls, data_bls, wgts={}, return_gains=False, gain_ants=[]):
+    """
+    Solve for the relative phase degeneracy between two polarizations of redundantly calibrated data.
+
+    Parameters:
+    -----------
+    model : DataContainer or RedDataContainer
+        Dictionary-like container mapping baselines to model visibilities
+    data : DataContainer or RedDataContainer
+        Dictionary-like container mapping baselines to data visibilities
+    model_bls : list of tuples
+        List of baseline tuples in model to use for calibration. Must correspond the same physical separations as data_bls.
+    data_bls : list of tuples
+        List of baseline tuples in data to use from calibration.
+    wgts : dict
+        Dictionary mapping baseline tuples to weights. Default is no weights.
+    return_gains : bool
+        If True, return gains instead of the relative phase difference between the two polarizations. 
+        Default is False.
+    gain_ants : list of tuples
+        List of antenna-pol tuples to return gains for. Default is None.
+
+    Returns:
+    --------
+    If return_gains is False:
+        delta : np.ndarray
+            Array of relative phase differences between the two polarizations. 
+    If return_gains is True:
+        delta_gains : dict
+            Dictionary mapping antenna keys like (0, 'Jee') to gains of the same shape of the data
+    """
+    # Create variable for the sum of the model and data product
+    summation = np.zeros_like(data[data_bls[0]])
+
+    for model_bl, data_bl in zip(model_bls, data_bls):
+
+        # Skip if the polarizations are the same
+        model_pol1, model_pol2 = utils.split_pol(model_bl[-1])
+        if model_pol1 == model_pol2:
+            continue
+
+        # Skip if the polarizations are the same
+        data_pol1, data_pol2 = utils.split_pol(data_bl[-1])
+        if model_pol1 == model_pol2:
+            continue
+
+        if data_pol1 < data_pol2:
```

Comment:
I find the use of < for strings to be very confusing and could result in difference answers depending on whether you use xx/yy or ee/nn (depending on one's x_orientation) but seems really unsatisfying. If you want to convert these to pyuvdata polarization numbers, I'd be fine with that.

Reply:
Alternatively, you could explicitly have the user specify which is the reference polarization (and therefore, effectively, which isn't)

Reply:
Settled on reference polarization